File: source/reference-ecommerce/stack/shopify-powered/lib/shopify-powered-stack.ts
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import * as apigateway from 'aws-cdk-lib/aws-apigateway';
import * as iam from 'aws-cdk-lib/aws-iam';
import * as path from 'path';

export class ShopifyPoweredStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // Create API Gateway
    const api = new apigateway.RestApi(this, 'ShopifyPoweredApi', {
      restApiName: 'Shopify Powered API',
      description: 'API for Shopify-powered e-commerce capabilities',
    });

    // Create Lambda functions
    const productManagementLambda = new lambda.Function(this, 'ProductManagementLambda', {
      runtime: lambda.Runtime.NODEJS_18_X,
      handler: 'index.handler',
      code: lambda.Code.fromAsset(path.join(__dirname, '../../..', 'integration/shopify-product-management')),
      environment: {
        SHOPIFY_SHOP_DOMAIN: process.env.SHOPIFY_SHOP_DOMAIN || '',
        SHOPIFY_ACCESS_TOKEN: process.env.SHOPIFY_ACCESS_TOKEN || '',
        SHOPIFY_API_VERSION: process.env.SHOPIFY_API_VERSION || '',
      },
    });

    const orderManagementLambda = new lambda.Function(this, 'OrderManagementLambda', {
      runtime: lambda.Runtime.NODEJS_18_X,
      handler: 'index.handler',
      code: lambda.Code.fromAsset(path.join(__dirname, '../../..', 'integration/shopify_order_management')),
      environment: {
        SHOPIFY_SHOP_DOMAIN: process.env.SHOPIFY_SHOP_DOMAIN || '',
        SHOPIFY_ACCESS_TOKEN: process.env.SHOPIFY_ACCESS_TOKEN || '',
        SHOPIFY_API_VERSION: process.env.SHOPIFY_API_VERSION || '',
      },
    });

    const cartManagementLambda = new lambda.Function(this, 'CartManagementLambda', {
      runtime: lambda.Runtime.NODEJS_18_X,
      handler: 'index.handler',
      code: lambda.Code.fromAsset(path.join(__dirname, '../../..', 'integration/elastic-path-cart')),
      environment: {
        ELASTIC_PATH_API_BASE_URL: process.env.ELASTIC_PATH_API_BASE_URL || '',
        ELASTIC_PATH_CLIENT_ID: process.env.ELASTIC_PATH_CLIENT_ID || '',
        ELASTIC_PATH_CLIENT_SECRET: process.env.ELASTIC_PATH_CLIENT_SECRET || '',
      },
    });

    // Create API Gateway resources and methods
    const productManagementResource = api.root.addResource('product-management');
    const productManagementProxy = productManagementResource.addProxy({
      defaultIntegration: new apigateway.LambdaIntegration(productManagementLambda),
      anyMethod: true,
    });

    const orderManagementResource = api.root.addResource('order-management');
    const orderManagementProxy = orderManagementResource.addProxy({
      defaultIntegration: new apigateway.LambdaIntegration(orderManagementLambda),
      anyMethod: true,
    });

    const cartManagementResource = api.root.addResource('cart-management');
    const cartManagementProxy = cartManagementResource.addProxy({
      defaultIntegration: new apigateway.LambdaIntegration(cartManagementLambda),
      anyMethod: true,
    });

    // Grant API Gateway permission to invoke Lambda functions
    productManagementLambda.grantInvoke(new iam.ServicePrincipal('apigateway.amazonaws.com'));
    orderManagementLambda.grantInvoke(new iam.ServicePrincipal('apigateway.amazonaws.com'));
    cartManagementLambda.grantInvoke(new iam.ServicePrincipal('apigateway.amazonaws.com'));

    // Output the API Gateway URL
    new cdk.CfnOutput(this, 'ApiUrl', {
      value: api.url,
      description: 'URL of the API Gateway',
    });
  }
}

File: converted-code/ShopifyPoweredStack.java
import software.amazon.awscdk.core.*;
import software.amazon.awscdk.services.lambda.*;
import software.amazon.awscdk.services.apigateway.*;
import software.amazon.awscdk.services.iam.*;

import java.util.Map;

public class ShopifyPoweredStack extends Stack {
    public ShopifyPoweredStack(final Construct scope, final String id) {
        this(scope, id, null);
    }

    public ShopifyPoweredStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        // Create API Gateway
        RestApi api = RestApi.Builder.create(this, "ShopifyPoweredApi")
                .restApiName("Shopify Powered API")
                .description("API for Shopify-powered e-commerce capabilities")
                .build();

        // Create Lambda functions
        Function productManagementLambda = Function.Builder.create(this, "ProductManagementLambda")
                .runtime(Runtime.NODEJS_18_X)
                .handler("index.handler")
                .code(Code.fromAsset("../../../integration/shopify-product-management"))
                .environment(Map.of(
                        "SHOPIFY_SHOP_DOMAIN", System.getenv("SHOPIFY_SHOP_DOMAIN") != null ? System.getenv("SHOPIFY_SHOP_DOMAIN") : "",
                        "SHOPIFY_ACCESS_TOKEN", System.getenv("SHOPIFY_ACCESS_TOKEN") != null ? System.getenv("SHOPIFY_ACCESS_TOKEN") : "",
                        "SHOPIFY_API_VERSION", System.getenv("SHOPIFY_API_VERSION") != null ? System.getenv("SHOPIFY_API_VERSION") : ""
                ))
                .build();

        Function orderManagementLambda = Function.Builder.create(this, "OrderManagementLambda")
                .runtime(Runtime.NODEJS_18_X)
                .handler("index.handler")
                .code(Code.fromAsset("../../../integration/shopify_order_management"))
                .environment(Map.of(
                        "SHOPIFY_SHOP_DOMAIN", System.getenv("SHOPIFY_SHOP_DOMAIN") != null ? System.getenv("SHOPIFY_SHOP_DOMAIN") : "",
                        "SHOPIFY_ACCESS_TOKEN", System.getenv("SHOPIFY_ACCESS_TOKEN") != null ? System.getenv("SHOPIFY_ACCESS_TOKEN") : "",
                        "SHOPIFY_API_VERSION", System.getenv("SHOPIFY_API_VERSION") != null ? System.getenv("SHOPIFY_API_VERSION") : ""
                ))
                .build();

        Function cartManagementLambda = Function.Builder.create(this, "CartManagementLambda")
                .runtime(Runtime.NODEJS_18_X)
                .handler("index.handler")
                .code(Code.fromAsset("../../../integration/elastic-path-cart"))
                .environment(Map.of(
                        "ELASTIC_PATH_API_BASE_URL", System.getenv("ELASTIC_PATH_API_BASE_URL") != null ? System.getenv("ELASTIC_PATH_API_BASE_URL") : "",
                        "ELASTIC_PATH_CLIENT_ID", System.getenv("ELASTIC_PATH_CLIENT_ID") != null ? System.getenv("ELASTIC_PATH_CLIENT_ID") : "",
                        "ELASTIC_PATH_CLIENT_SECRET", System.getenv("ELASTIC_PATH_CLIENT_SECRET") != null ? System.getenv("ELASTIC_PATH_CLIENT_SECRET") : ""
                ))
                .build();

        // Create API Gateway resources and methods
        Resource productManagementResource = api.getRoot().addResource("product-management");
        productManagementResource.addProxy(ProxyResourceOptions.builder()
                .defaultIntegration(new LambdaIntegration(productManagementLambda))
                .anyMethod(true)
                .build());

        Resource orderManagementResource = api.getRoot().addResource("order-management");
        orderManagementResource.addProxy(ProxyResourceOptions.builder()
                .defaultIntegration(new LambdaIntegration(orderManagementLambda))
                .anyMethod(true)
                .build());

        Resource cartManagementResource = api.getRoot().addResource("cart-management");
        cartManagementResource.addProxy(ProxyResourceOptions.builder()
                .defaultIntegration(new LambdaIntegration(cartManagementLambda))
                .anyMethod(true)
                .build());

        // Grant API Gateway permission to invoke Lambda functions
        productManagementLambda.grantInvoke(new ServicePrincipal("apigateway.amazonaws.com"));
        orderManagementLambda.grantInvoke(new ServicePrincipal("apigateway.amazonaws.com"));
        cartManagementLambda.grantInvoke(new ServicePrincipal("apigateway.amazonaws.com"));

        // Output the API Gateway URL
        new CfnOutput(this, "ApiUrl", CfnOutputProps.builder()
                .value(api.getUrl())
                .description("URL of the API Gateway")
                .build());
    }
}